src/transmission.c

```diff
@@ -0,0 +1,48 @@
+#include <freertos/FreeRTOS.h>
+#include <stdint.h>
+#include <unistd.h>
+
+#include <lwip/err.h>
+#include <lwip/udp.h>
+#include <lwip/sockets.h>
+
+#include "transmission.h"
+
+#define SERVER_IP "192.168.100.1"
+#define PORT 4321
+
+struct sockaddr_in dest_addr;
+int32_t sock = -1;
+
+void
+transmission_init(void)
+{
+    uint8_t addr_family = AF_INET;
+    char address[32];
+
+    dest_addr.sin_addr.s_addr = inet_addr(SERVER_IP);
+    dest_addr.sin_family = addr_family;
+    dest_addr.sin_port = htons(PORT);
+    inet_ntoa_r(dest_addr.sin_addr, (char *)address, sizeof(address) - 1);
+
+    while (sock < 0)
+    {
+        sock = socket(addr_family, SOCK_DGRAM, IPPROTO_IP);
+    }
```

Comment:
why is this a busy-wait?

Reply:
Sometimes  socket initialisation can fail because of ENOM or similar problems. There is no error handling in this code so far. Will add a comment